test/e2e/e2e_helpers_test.go

```diff
@@ -424,6 +423,65 @@ func sortNamespacedNames(nns []types.NamespacedName) {
 }
 
 func testDataPath(description ginkgo.GinkgoTestDescription) string {
-       path := filepath.Join(description.ComponentTexts...)
-       return strings.ReplaceAll(path, " ", "_")
+	path := filepath.Join(description.ComponentTexts...)
+	return strings.ReplaceAll(path, " ", "_")
+}
+
+func GetIstioObject(istio *istiov1beta1.Istio, namespace, name string) error {
+	return testEnv.Client.Get(context.TODO(), client.ObjectKey{
+		Namespace: namespace,
+		Name:      name},
+		istio)
+}
+
+func GetMixerlessTelemetryStatus(istio *istiov1beta1.Istio, namespace, name string) (string, error) {
+
+	err := GetIstioObject(istio, namespace, name)
+	if err != nil {
+		return "", err
+	}
+	// query current state and return it
+	if istio.Spec.MixerlessTelemetry.Enabled == nil {
+		return "N", nil
+	}
+	if *istio.Spec.MixerlessTelemetry.Enabled {
+		return "T", nil
+	}
+	return "F", nil
```

Comment:
I don't think the "N", "T", "F" stuff helps. It should simply be:

```suggestion
	return istio.Spec.MixerlessTelemetry.Enabled, nil
```

Reply:
This has been cleaned up without the Strings, and only uses a *bool